### DISCUSSION TITLE
Dynamic routes based on global value

### DISCUSSION THREAD

**Discussion by royvanv** [statamic/cms#10222] [Q&A]
Before submitting a bug report I want to see if there's another solution for my specific situation.

I have a Statamic  website with Jetstream installed and I have dynamic routes for the terms and privacy pages. The page IDs are stored in a global and the routes are created for the page's slug within `routes/jetstream.php`.

I recently started the update to Statamic 5 and Laravel 11, but broke this code. It now throws a `Symfony\Component\Routing\Exception\RouteNotFoundException` with the message "Route [statamic.cp.collections.entries.edit] not defined.". Which seems logical to me, because `routes/jetstream.php` is loaded before the Statamic CP routes are loaded.

My current work-around is to temporarily disable the CP, add the routes and re-enable the CP again.

Does anyone have a suggestion to improve this?
___

Excerpt of `routes/jetstream.php`:
```php
<?php

use App\Helpers\Statamic;
use Illuminate\Support\Facades\Route;
use Laravel\Jetstream\Jetstream;
use Illuminate\Support\Facades\Config;

Route::group(['middleware' => config('jetstream.middleware', ['web'])], function ()
{
    if (Jetstream::hasTermsAndPrivacyPolicyFeature())
    {
        // Statamic CP routes are not set yet and cause an error.
        if ($cpEnabled = Config::get('statamic.cp.enabled'))
        {
            Config::set('statamic.cp.enabled', false);
        }

        Statamic::routeForEntry('pages', 'terms')->name('terms.show');
        Statamic::routeForEntry('pages', 'privacy')->name('policy.show');

        if ($cpEnabled)
        {
            Config::set('statamic.cp.enabled', true);
        }
    }

    // ...
});
```

Simplified version of my helper `App\Helpers\Statamic`:
```php
<?php

namespace App\Helpers;

use Illuminate\Routing\Route;
use Illuminate\Support\Facades\Route as RouteFacade;
use Statamic\Entries\Entry;
use Statamic\Facades\GlobalSet;
use Statamic\Http\Controllers\FrontendController;

final class Statamic
{
    public static function routeForEntry(string $set, string $key, array | string $verbs = 'get'): Route
    {
        $entryId = GlobalSet::findByHandle($set)?->in('default')?->data()?->get($key);
        $entry = Entry::find($entryId);

        return RouteFacade::match($verbs, '/' . $entry->slug(), [FrontendController::class, 'index']);
    }
}
```

**Comment by edalzell** [comment]
Can you put that code in a statamic booted callback?

**Reply by royvanv** [reply]
Do you mean calling the `booted` method in `\Statamic\Statamic`?

I tried using that in the `boot` method of my service provider, but the error remains.
